x/finality/keeper/tallying.go

```diff
@@ -33,6 +33,9 @@ func (k Keeper) TallyBlocks(ctx context.Context, maxFinalizedBlocks uint64) {
 	}
 
 	currentLastBlockHeight := uint64(sdkCtx.HeaderInfo().Height)
+	if !k.HasBlock(ctx, currentLastBlockHeight) {
```

Comment:
Now that the `IndexBlock` is called only here, the if condition will be always true, no?

Reply:
true, but this a bit defensive move to avoid similar situation in the future i.e somebody using `TallyBlocks` on unindexed block

Reply:
yes, but I still would like to keep this check to avoid setting a block to unfinalized again somehow

Reply:
ok, lgtm! :+1: